tests/MongoDbTest.php

```diff
@@ -42,8 +42,13 @@ public function it_can_generate_a_dump_command_with_compression_enabled()
             ->enableCompression()
             ->getDumpCommand('dbname.gz');
 
-        $this->assertSame('\'mongodump\' --db dbname'
-            .' --archive --host localhost --port 27017 | gzip > "dbname.gz"', $dumpCommand);
+        $this->assertSame('if output=$(\'mongodump\' --db dbname --archive --host localhost --port 27017);
```

Comment:
Let's add a function `removeErrorHandling` that removes that `if` until `fi` part so we don't have to repeat that in multiple tests

Reply:
Don't see this as necessary, I'd rather just edit the tests. If we use compression, we need the error handling.

The following test always tests the path (it_can_generate_a_dump_command_with_absolute_path_having_space_and_brackets), I'd just remove the compression call there so the output is clean and only testing the path.

Also, these 2 tests are the exact same, the first one is just testing a deprecated function
```
it_can_generate_a_dump_command_with_compression_enabled
it_can_generate_a_dump_command_with_gzip_compressor_enabled
```

Reply:
Ok, let's leave it like that